base/commands/migration/status_stages.go

```diff
@@ -0,0 +1,152 @@
+	"github.com/hazelcast/hazelcast-go-client"
+	"github.com/hazelcast/hazelcast-go-client/types"
+)
+
+type StatusStages struct {
+	migrationID              string
+	ci                       *hazelcast.ClientInternal
+	migrationsInProgressList *hazelcast.List
+	statusMap                *hazelcast.Map
+	updateTopic              *hazelcast.Topic
+	topicListenerID          types.UUID
+	updateMsgChan            chan UpdateMessage
+}
+
+func NewStatusStages() *StatusStages {
+	return &StatusStages{}
+}
+
+func (st *StatusStages) Build(ctx context.Context, ec plug.ExecContext) []stage.Stage {
+	return []stage.Stage{
+		{
+			ProgressMsg: "Connecting to the migration cluster",
+			SuccessMsg:  "Connected to the migration cluster",
+			FailureMsg:  "Could not connect to the migration cluster",
+			Func:        st.connectStage(ctx, ec),
+		},
+		{
+			ProgressMsg: "Fetching migration status",
+			SuccessMsg:  "Fetched migration status",
+			FailureMsg:  "Could not fetch migration status",
+			Func:        st.fetchStage(ctx, ec),
+		},
+	}
+}
+
+type MigrationInProgress struct {
+	MigrationID string `json:"migrationId"`
+}
+
+func (st *StatusStages) connectStage(ctx context.Context, ec plug.ExecContext) func(statuser stage.Statuser) error {
+	return func(status stage.Statuser) error {
+		var err error
+		st.ci, err = ec.ClientInternal(ctx)
+		if err != nil {
+			return err
+		}
+		st.migrationsInProgressList, err = st.ci.Client().GetList(ctx, MigrationsInProgressList)
+		if err != nil {
+			return err
+		}
+		all, err := st.migrationsInProgressList.GetAll(ctx)
+		if err != nil {
+			return err
+		}
+		m := all[0].(MigrationInProgress)
+		st.migrationID = m.MigrationID
+		st.statusMap, err = st.ci.Client().GetMap(ctx, MakeStatusMapName(st.migrationID))
+		if err != nil {
+			return err
+		}
+		st.updateTopic, err = st.ci.Client().GetTopic(ctx, MakeUpdateTopicName(st.migrationID))
+		if err != nil {
+			return err
+		}
+		st.updateMsgChan = make(chan UpdateMessage)
+		_, err = st.updateTopic.AddMessageListener(ctx, st.topicListener)
+		return err
+	}
+}
+
+func (st *StatusStages) fetchStage(ctx context.Context, ec plug.ExecContext) func(statuser stage.Statuser) error {
+	return func(stage.Statuser) error {
+		defer st.updateTopic.RemoveListener(ctx, st.topicListenerID)
+		for {
+			select {
+			case msg := <-st.updateMsgChan:
+				ms, err := readMigrationStatus(ctx, st.statusMap)
+				if err != nil {
+					return fmt.Errorf("reading status: %w", err)
+				}
+				ec.PrintlnUnnecessary(msg.Message)
+				if slices.Contains([]Status{StatusComplete, StatusFailed, StatusCanceled}, msg.Status) {
+					ec.PrintlnUnnecessary(fmt.Sprintf("Completion Percentage: %f", ms.CompletionPercentage))
+					ec.PrintlnUnnecessary(ms.Report)
+					if len(ms.Errors) > 0 {
+						ec.PrintlnUnnecessary(fmt.Sprintf("migration failed with following error(s):\n%s", strings.Join(ms.Errors, "\n")))
+					}
+					if len(ms.Migrations) > 0 {
+						var rows []output.Row
```

Comment:
🤔 What is this "rows" object for ?

Reply:
Just to print migrations as a table.

Reply:
I think if the migration is complete, you could just fetch the migration report and print it.

Reply:
https://github.com/hazelcast/hazelcast-commandline-client/pull/367/commits/d5591a030703633a6e871e9362197e17f2a20a79